### PR TITLE
Fixed issue #14400: Unable to use self ot that fixed string in Expression

### PR DESCRIPTION
**Pull request by Shnoulle**
Fixed issue #14400: Unable to use self or that fixed string in Expression
Dev: add self,that and this in reserved var for Question rules 
Fixed issue : don't replace title starting with self or that in expression

### CODE REVIEW

**Comment by Shnoulle** [comment]
Sill have issue with question title as `selfToto`

**Comment by Shnoulle** [comment]
CodeFactor is unhappy :/ But in general : CodeFactor didn't like EM …

File: application/models/Question.php [hunk 1]
```diff
@@ -171,13 +171,10 @@ public function rules()
...
-                    'SID',
-                    'QID',
-                    'GID',
-                    'SAVEDID',
-                    'SGQ',
-                    'TOKEN',
+                    'LANG','SID', // Global var
+                    'SAVEDID','TOKEN', // current survey related var
+                    'QID','GID','SGQ', // current question related var
+                    'self','that','this', // EM reserved variables
```
Comment (maziminke) [inline application/models/Question.php]: Does that mean that "self" as an answer code will not be accepted anymore?

Comment (Shnoulle) [reply application/models/Question.php]: As a question title , not answer or Subquestion title

**Comment by Shnoulle** [comment]
@maziminke : i think it's OK now BUT : i like more testing with this fix. Since current 3.15 is old … think i wait next release to merge (and then have a week to test)

File: application/models/Question.php (hunk 1 above)
Comment (maziminke) [reply application/models/Question.php]: Ahh, that makes sense. Being able to use it as answer text would be very helpful so if possible let's keep that.

Comment (Shnoulle) [reply application/models/Question.php]: Yes, no issue in answer text : EM have issue with it currently, but it's an issue in EM.

But clearly : we must not allow it in question title/code

**Pull request merged**